app/models/user.php

```diff
@@ -108,27 +106,81 @@ public function login()
         }
 
         $_SESSION['user_id'] = $user['id'];
-        $_SESSION['username'] =$user['username'];
+        $_SESSION['first_name'] =$user['first_name'];       
     }
 
-    public function is_password_match()
+    public function isPasswordMatch()
     {
         return $this->password == $this->confirm_password;
     }
 
-    public function is_username_exist()
+    public function isUsernameExist()
     {
         $db = DB::conn();
-        $username_exist = $db->row("SELECT username FROM user WHERE username = ?", array($this->username));
+        $username_exist = $db->row("SELECT username FROM user WHERE 
+                                    username = ?", array($this->username));
 
         return !$username_exist;
     }
 
-       public function is_email_exist()
+    public function isEmailExist()
     {
         $db = DB::conn();
-        $username_exist = $db->row("SELECT email FROM user WHERE email = ?", array($this->email));
+        $params = array($this->email, 
+                        $_SESSION['user_id']
+                    );
 
-        return !$username_exist;
+        $email_exist = $db->row("SELECT email FROM user WHERE
+                                 email = ? && id != ?", $params);
+
+        return !$email_exist;
+    }
+
+    public static function getUsername($user_id)
+    {
+        $db = DB::conn();
+        $user = $db->row("SELECT username FROM user WHERE
+                            id = ?", array($user_id));    
```

Comment:
(low priority) I think this is short enough to be put in 1-line, but if not, I think it's better to put the WHERE clause in the next line to make it more easy to read :bow: 
```
$user = $db->row("SELECT username FROM user
                               WHERE id =  ?", array($user_id));
```